### ISSUE TITLE
TSLM refactoring

### ISSUE THREAD

**Issue by fwaldmann** [lmlib/lmlib#7]
Two Segmented Line Model (TSLM) restructure

Suggested Coding

```
cost = TSLM.create_cost(ab=(-20, 20), gs=(50, 50))
H_horizontal = TSLM.H_horizontal (read only property)
```

TSLM(ABC) abstract class

place: statespace.applications.TSLM

**Comment by fwaldmann** [comment]
first version implemented in branch:dev

**Comment by fwaldmann** [comment]
Fixed in release v2.1.0
rev. 841cf96ec478daacf2fdbf03f14607f5151ec5ec
